### ISSUE TITLE
Swap marked as recoverable is not recoverable

### ISSUE THREAD

**Issue by naezith** [KomodoPlatform/atomicDEX-Pro#182]
I have this swap, `is_recoverable` field is true so it shows `RECOVER FUNDS` button.

![image](https://user-images.githubusercontent.com/6732486/82671078-ea411900-9c46-11ea-8b8d-f2d8ed1f3f84.png)

In details it says `MakerPaymentWaitConfirmFailed`

![image](https://user-images.githubusercontent.com/6732486/82671029-d990a300-9c46-11ea-89c5-f974a04ad616.png)

`Error Log` there is this:

```taker_swap:825] !wait for maker payment confirmations: rpc_clients:108] Waited too long until 1590104402 for transaction Transaction { version: 4, n_time: None, overwintered: true, version_group_id: 2301567109, inputs: [TransactionInput { previous_output: OutPoint { hash: ffb0f08d39c631989dbfa53558cfa4a457b14578c1c843d54780b6ded255c71c, index: 1 }, script_sig: 4730440220484651b2f34040346a45946ddf0e540184fab7d4697c30f9f52a6fd1344341f502202b08dc6b734a65656babae85cf1383835c6d082e6e6e6cbbc3fb749b55b4e3d1012103683c77e807a47dcd559fa60a6510087e5c5aa0016c094cf5eb4d7e002db18e9f, sequence: 4294967295, script_witness: [] }], outputs: [TransactionOutput { value: 3306666, script_pubkey: a91473a0f576bf4e51f506faedd668c155e955c7a12087 }, TransactionOutput { value: 566890723, script_pubkey: 76a914aa8fc9fcbf345510d419ca273a6e9897b488ee1688ac }], lock_time: 1590101367, expiry_height: 0, shielded_spends: [], shielded_outputs: [], join_splits: [], value_balance: 0, join_split_pubkey: 0000000000000000000000000000000000000000000000000000000000000000, join_split_sig: 00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000, binding_sig: 00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000, zcash: true, str_d_zeel: None } to be confirmed 2 times```

Here is the full data of that entry:

![image](https://user-images.githubusercontent.com/6732486/82670986-c978c380-9c46-11ea-98df-260dca6c4544.png)

When `RECOVER FUNDS` button is pressed, this error appears:

```
{
    "error": "rpc:277] lp_swap:763] lp_swap:477] taker_swap:1150] Taker payment is not found, swap is not recoverable"
}
```

Here is the ID: `24c3b70b-cab9-490c-8edb-d95fcf198e7d`

Full data:
```json
{
    "error_events": [
      "StartFailed",
      "NegotiateFailed",
      "TakerFeeSendFailed",
      "MakerPaymentValidateFailed",
      "MakerPaymentWaitConfirmFailed",
      "TakerPaymentTransactionFailed",
      "TakerPaymentWaitConfirmFailed",
      "TakerPaymentDataSendFailed",
      "TakerPaymentWaitForSpendFailed",
      "MakerPaymentSpendFailed",
      "TakerPaymentWaitRefundStarted",
      "TakerPaymentRefunded",
      "TakerPaymentRefundFailed"
    ],
    "events": [
      {
        "data": {
          "lock_duration": 7800,
          "maker": "683c77e807a47dcd559fa60a6510087e5c5aa0016c094cf5eb4d7e002db18e9f",
          "maker_amount": "0.03306666666666666666666666666666666666666666666666666666666666666666666666666666666666666666666666667",
          "maker_coin": "DEX",
          "maker_coin_start_block": 1132583,
          "maker_payment_confirmations": 2,
          "maker_payment_requires_nota": true,
          "maker_payment_wait": 1590104402,
          "my_persistent_pub": "03f813f322d82167027f635f08686fdc615f2f98d8f5f49091dafc62d520666aa5",
          "started_at": 1590101282,
          "taker_amount": "0.496",
          "taker_coin": "KMD",
          "taker_coin_start_block": 1887916,
          "taker_payment_confirmations": 2,
          "taker_payment_lock": 1590109082,
          "taker_payment_requires_nota": true,
          "uuid": "24c3b70b-cab9-490c-8edb-d95fcf198e7d"
        },
        "human_timestamp": "2020-05-21    22:48:02.551",
        "state": "Started",
        "timestamp": 1590101282551
      },
      {
        "data": {
          "maker_payment_locktime": 1590116883,
          "maker_pubkey": "03683c77e807a47dcd559fa60a6510087e5c5aa0016c094cf5eb4d7e002db18e9f",
          "secret_hash": "3c79aff4e91435d57241dd9702813fe6d1f63108"
        },
        "human_timestamp": "2020-05-21    22:49:03.769",
        "state": "Negotiated",
        "timestamp": 1590101343769
      },
      {
        "data": {
          "block_height": 0,
          "coin": "KMD",
          "fee_details": {
            "amount": "0.00001"
          },
          "from": [
            "RH8WgfYXbeMgF96vCqfKo47TkFApNXkQM2"
          ],
          "internal_id": "9f0d5aa87e4155d45f4852560d7f61d136bd1a8a74c74c3353335ced19e5966a",
          "my_balance_change": "-0.00058451",
          "received_by_me": "2.49703889",
          "spent_by_me": "2.4976234",
          "timestamp": 0,
          "to": [
            "RH8WgfYXbeMgF96vCqfKo47TkFApNXkQM2",
            "RThtXup6Zo7LZAi8kRWgjAyi1s4u6U9Cpf"
          ],
          "total_amount": "2.4976234",
          "tx_hash": "9f0d5aa87e4155d45f4852560d7f61d136bd1a8a74c74c3353335ced19e5966a",
          "tx_hex": "0400008085202f8901ae23b8bc9466c8cd9c243613e52c46698f59676deca140d93c70f5416825f416010000006b483045022100f4fd9d5d4c1f63b3d30a124c0da28d377499c2d7bf57f6cdf3b7473e048acaf50220292440797f6f1c327e08fb0ca4ee6fd0eb8e037ea144bd7a315bceafc45c72da012103f813f322d82167027f635f08686fdc615f2f98d8f5f49091dafc62d520666aa5ffffffff026be00000000000001976a914ca1e04745e8ca0c60d8c5881531d51bec470743f88acd12de20e000000001976a914561ccb4cb9159a1ba1b81c133507ea950e065edc88ac61fdc65e000000000000000000000000000000"
        },
        "human_timestamp": "2020-05-21    22:49:07.242",
        "state": "TakerFeeSent",
        "timestamp": 1590101347242
      },
      {
        "data": {
          "block_height": 1132584,
          "coin": "DEX",
          "fee_details": {
            "amount": "0.00001"
          },
          "from": [
            "RQq3CmCy7DzwSbT1RPZDfVPcC3RAjyR8DJ"
          ],
          "internal_id": "8b324f60d68cdc8c5f75772a0924998b0c0f8bbd4404612bf918ecab76e39b2d",
          "my_balance_change": "0",
          "received_by_me": "0",
          "spent_by_me": "0",
          "timestamp": 1590101376,
          "to": [
            "RQq3CmCy7DzwSbT1RPZDfVPcC3RAjyR8DJ",
            "bPGfDubsFzzQ5jctEfsqzhwJBdiwUTLruM"
          ],
          "total_amount": "5.70198389",
          "tx_hash": "8b324f60d68cdc8c5f75772a0924998b0c0f8bbd4404612bf918ecab76e39b2d",
          "tx_hex": "0400008085202f8901ffb0f08d39c631989dbfa53558cfa4a457b14578c1c843d54780b6ded255c71c010000006a4730440220484651b2f34040346a45946ddf0e540184fab7d4697c30f9f52a6fd1344341f502202b08dc6b734a65656babae85cf1383835c6d082e6e6e6cbbc3fb749b55b4e3d1012103683c77e807a47dcd559fa60a6510087e5c5aa0016c094cf5eb4d7e002db18e9fffffffff02aa7432000000000017a91473a0f576bf4e51f506faedd668c155e955c7a12087e310ca21000000001976a914aa8fc9fcbf345510d419ca273a6e9897b488ee1688ac7705c75e000000000000000000000000000000"
        },
        "human_timestamp": "2020-05-21    22:49:47.934",
        "state": "MakerPaymentReceived",
        "timestamp": 1590101387934
      },
      {
        "human_timestamp": "2020-05-21    22:49:47.935",
        "state": "MakerPaymentWaitConfirmStarted",
        "timestamp": 1590101387935
      },
      {
        "data": {
          "error": "taker_swap:825] !wait for maker payment confirmations: rpc_clients:108] Waited too long until 1590104402 for transaction Transaction { version: 4, n_time: None, overwintered: true, version_group_id: 2301567109, inputs: [TransactionInput { previous_output: OutPoint { hash: ffb0f08d39c631989dbfa53558cfa4a457b14578c1c843d54780b6ded255c71c, index: 1 }, script_sig: 4730440220484651b2f34040346a45946ddf0e540184fab7d4697c30f9f52a6fd1344341f502202b08dc6b734a65656babae85cf1383835c6d082e6e6e6cbbc3fb749b55b4e3d1012103683c77e807a47dcd559fa60a6510087e5c5aa0016c094cf5eb4d7e002db18e9f, sequence: 4294967295, script_witness: [] }], outputs: [TransactionOutput { value: 3306666, script_pubkey: a91473a0f576bf4e51f506faedd668c155e955c7a12087 }, TransactionOutput { value: 566890723, script_pubkey: 76a914aa8fc9fcbf345510d419ca273a6e9897b488ee1688ac }], lock_time: 1590101367, expiry_height: 0, shielded_spends: [], shielded_outputs: [], join_splits: [], value_balance: 0, join_split_pubkey: 0000000000000000000000000000000000000000000000000000000000000000, join_split_sig: 00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000, binding_sig: 00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000, zcash: true, str_d_zeel: None } to be confirmed 2 times"
        },
        "human_timestamp": "2020-05-22    09:14:29.695",
        "state": "MakerPaymentWaitConfirmFailed",
        "timestamp": 1590138869695
      },
      {
        "human_timestamp": "2020-05-22    09:14:29.696",
        "state": "Finished",
        "timestamp": 1590138869696
      }
    ],
    "is_recoverable": true,
    "maker_amount": "0.03306667",
    "maker_coin": "DEX",
    "my_info": {
      "my_amount": "0.496",
      "my_coin": "KMD",
      "other_amount": "0.03306667",
      "other_coin": "DEX",
      "started_at": 1590101282
    },
    "success_events": [
      "Started",
      "Negotiated",
      "TakerFeeSent",
      "MakerPaymentReceived",
      "MakerPaymentWaitConfirmStarted",
      "MakerPaymentValidatedAndConfirmed",
      "TakerPaymentSent",
      "TakerPaymentSpent",
      "MakerPaymentSpent",
      "Finished"
    ],
    "taker_amount": "0.496",
    "taker_coin": "KMD",
    "type": "Taker",
    "uuid": "24c3b70b-cab9-490c-8edb-d95fcf198e7d",
    "is_recent_swap": true
  }
```

**Comment by artemii235** [comment]
The swap is `recoverable` because `TakerPaymentSent` is missing. It is possible that coin RPC returns the error but payment was broadcast in fact. So it is expected behavior that swap is recoverable, but recover attempt fails because payment is not found. We've discussed this case some time ago in context of mobile GUI, the idea was to return the `success` response in case payment is not found or add `recoverable_again` field to let GUIs know whether additional recovery attempts make sense.

Also I've noticed that in `MakerPaymentWaitConfirmFailed` case swap is 100% *not* recoverable because MM2 doesn't make attempt to send the taker payment in such case, will fix.
